src/watcher.js

```diff
@@ -331,14 +331,25 @@ Watcher.prototype.teardown = function () {
  * @param {*} val
  */
 
-function traverse (val) {
+function traverse (val, walkedObjs) {
   var i, keys
+
+  !walkedObjs && (walkedObjs = {})
```

Comment:
I'm curious about this notation. How is it different from `walkedObjs = walkedObjs || {}` ?